File: src/Entity/Event.php
<?php

namespace App\Entity;

use App\Repository\EventRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;




#[ORM\Entity(repositoryClass: EventRepository::class)]
class Event
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id_event = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "champ null")]
    private ?string $Titre_event = null;

    #[ORM\Column(type: Types::DECIMAL, precision: 10, scale: '0')]
    #[Assert\NotBlank(message: "champ null")]
    #[Assert\Positive(message: "Number should be positive")]
    private ?string $Prix_event = null;

    #[ORM\Column(type: Types::DECIMAL, precision: 10, scale: '0')]
    #[Assert\NotBlank(message: "champ null")]
    #[Assert\Positive(message: "Number should be positive")]
    private ?string $Nbr_personnes = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "champ null")]
    private ?string $Description_event = null;

    #[ORM\Column(length: 255)]
    private ?\DateTimeImmutable $Date_event = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "champ null")]
    private ?string $Etat_event = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "champ null")]
    private ?string $Lieu_event = null;

    #[ORM\ManyToOne(inversedBy: 'events')]
    #[ORM\JoinColumn(nullable: false, referencedColumnName:"id_category",onDelete:"CASCADE")]
    private ?Category $Categorie_Event = null;

    #[ORM\OneToMany(targetEntity: Reservation::class, mappedBy: 'Event')]
    private Collection $reservations;

    public function __construct()
    {
        $this->reservations = new ArrayCollection();
    }



    public function getIdEvent(): ?int
    {
        return $this->id_event;
    }

    public function getTitreEvent(): ?string
    {
        return $this->Titre_event;
    }

    public function setTitreEvent(string $Titre_event): static
    {
        $this->Titre_event = $Titre_event;

        return $this;
    }

    public function getPrixEvent(): ?string
    {
        return $this->Prix_event;
    }

    public function setPrixEvent(string $Prix_event): static
    {
        $this->Prix_event = $Prix_event;

        return $this;
    }

    public function getNbrPersonnes(): ?string
    {
        return $this->Nbr_personnes;
    }

    public function setNbrPersonnes(string $Nbr_personnes): static
    {
        $this->Nbr_personnes = $Nbr_personnes;

        return $this;
    }

    public function getDescriptionEvent(): ?string
    {
        return $this->Description_event;
    }

    public function setDescriptionEvent(string $Description_event): static
    {
        $this->Description_event = $Description_event;

        return $this;
    }

    public function getDateEvent(): ?\DateTimeImmutable
    {
        return $this->Date_event;
    }

    public function setDateEvent(\DateTimeImmutable $Date_event): static
    {
        $this->Date_event = $Date_event;

        return $this;
    }

    public function getEtatEvent(): ?string
    {
        return $this->Etat_event;
    }

    public function setEtatEvent(string $Etat_event): static
    {
        $this->Etat_event = $Etat_event;

        return $this;
    }

    public function getLieuEvent(): ?string
    {
        return $this->Lieu_event;
    }

    public function setLieuEvent(string $Lieu_event): static
    {
        $this->Lieu_event = $Lieu_event;

        return $this;
    }

    public function getCategorieEvent(): ?Category
    {
        return $this->Categorie_Event;
    }

    public function setCategorieEvent(?Category $Categorie_Event): static
    {
        $this->Categorie_Event = $Categorie_Event;

        return $this;
    }



    /**
     * @return Collection<int, Reservation>
     */
    public function getReservations(): Collection
    {
        return $this->reservations;
    }

    public function addReservation(Reservation $reservation): static
    {
        if (!$this->reservations->contains($reservation)) {
            $this->reservations->add($reservation);
            $reservation->setEvent($this);
        }

        return $this;
    }

    public function removeReservation(Reservation $reservation): static
    {
        if ($this->reservations->removeElement($reservation)) {
            // set the owning side to null (unless already changed)
            if ($reservation->getEvent() === $this) {
                $reservation->setEvent(null);
            }
        }

        return $this;
    }


    public function __toString()
    {
        return $this->Titre_event; // ou tout autre champ que vous voulez afficher
    }



}

File: src/main/java/models/Event.java
package models;

import java.sql.Date;
import java.util.Objects;

public class Event {
    private int id_event;
    private String titre_event;
    private String description_event;
    private String etat_event;
    private String lieu_event;
    private float prix_event;
    private float nbr_personnes;
    private Date date_event;
    private int categorie_event_id;

    public Event(int id_event, String titre_event, String description_event, String etat_event, String lieu_event, float prix_event, float nbr_personnes, Date date_event, int categorie_event_id) {
        this.id_event = id_event;
        this.titre_event = titre_event;
        this.description_event = description_event;
        this.etat_event = etat_event;
        this.lieu_event = lieu_event;
        this.prix_event = prix_event;
        this.nbr_personnes = nbr_personnes;
        this.date_event = date_event;
        this.categorie_event_id = categorie_event_id;
    }

    public Event(String titre_event, String description_event, String etat_event, String lieu_event, float prix_event, float nbr_personnes, Date date_event, int categorie_event_id) {
        this.titre_event = titre_event;
        this.description_event = description_event;
        this.etat_event = etat_event;
        this.lieu_event = lieu_event;
        this.prix_event = prix_event;
        this.nbr_personnes = nbr_personnes;
        this.date_event = date_event;
        this.categorie_event_id = categorie_event_id;
    }

    public Event() {
    }

    public int getId_event() {
        return id_event;
    }

    public void setId_event(int id_event) {
        this.id_event = id_event;
    }

    public String getTitre_event() {
        return titre_event;
    }

    public void setTitre_event(String titre_event) {
        this.titre_event = titre_event;
    }

    public String getDescription_event() {
        return description_event;
    }

    public void setDescription_event(String description_event) {
        this.description_event = description_event;
    }

    public String getEtat_event() {
        return etat_event;
    }

    public void setEtat_event(String etat_event) {
        this.etat_event = etat_event;
    }

    public String getLieu_event() {
        return lieu_event;
    }

    public void setLieu_event(String lieu_event) {
        this.lieu_event = lieu_event;
    }

    public float getPrix_event() {
        return prix_event;
    }

    public void setPrix_event(float prix_event) {
        this.prix_event = prix_event;
    }

    public float getNbr_personnes() {
        return nbr_personnes;
    }

    public void setNbr_personnes(float nbr_personnes) {
        this.nbr_personnes = nbr_personnes;
    }

    public Date getDate_event() {
        return date_event;
    }

    public void setDate_event(Date date_event) {
        this.date_event = date_event;
    }

    public int getCategorie_event_id() {
        return categorie_event_id;
    }

    public void setCategorie_event_id(int categorie_event_id) {
        this.categorie_event_id = categorie_event_id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Event event = (Event) o;
        return id_event == event.id_event &&
                Float.compare(prix_event, event.prix_event) == 0 &&
                Float.compare(nbr_personnes, event.nbr_personnes) == 0 &&
                date_event.compareTo(event.date_event) == 0 &&
                categorie_event_id == event.categorie_event_id &&
                Objects.equals(titre_event, event.titre_event) &&
                Objects.equals(description_event, event.description_event) &&
                Objects.equals(etat_event, event.etat_event) &&
                Objects.equals(lieu_event, event.lieu_event);
    }



    @Override
    public int hashCode() {
        return Objects.hash(id_event, titre_event, description_event, etat_event, lieu_event, prix_event, nbr_personnes, date_event, categorie_event_id);
    }

    @Override
    public String toString() {
        return "Event{" +
                "id_event=" + id_event +
                ", titre_event='" + titre_event + '\'' +
                ", description_event='" + description_event + '\'' +
                ", etat_event='" + etat_event + '\'' +
                ", lieu_event='" + lieu_event + '\'' +
                ", prix_event=" + prix_event +
                ", nbr_personnes=" + nbr_personnes +
                ", date_event=" + date_event +
                ", categorie_event_id=" + categorie_event_id +
                '}';
    }
}
